test/common/network/address_impl_test.cc

```diff
@@ -317,6 +321,18 @@ TEST(PipeInstanceTest, Basic) {
   EXPECT_EQ("/foo", address.asString());
   EXPECT_EQ(Type::Pipe, address.type());
   EXPECT_EQ(nullptr, address.ip());
+  EXPECT_EQ(nullptr, address.envoyInternalAddress());
+}
+
+TEST(InteralInstanceTest, Basic) {
+  EnvoyInternalInstance address("listener_foo");
+  EXPECT_EQ("envoy://listener_foo", address.asString());
```

Comment:
Move @alyssawilk 's discussion to this position

I wonder if we should have the string representation be envoy:// in config.
@htuch any thoughts here?

I could imagine for instance someone naming their listener after the address it is listening on, and there otherwise being ambiguity.